pkg/ebpf/c/tracee.bpf.c

```diff
@@ -4819,6 +4825,59 @@ int BPF_KPROBE(trace_ret_do_splice)
     return events_perf_submit(&data, DIRTY_PIPE_SPLICE, 0);
 }
 
+static __always_inline void send_file_operations_struct(event_data_t *data, struct file * called_file)
+{
+    struct inode *file_inode = get_inode_from_file(called_file);
+    struct file_operations *fops = (struct file_operations *)READ_KERN(file_inode->i_fop);
+    if (fops == NULL)
+        return ;
+
+    unsigned long iterate_shared_addr = (unsigned long) READ_KERN(fops->iterate_shared);
+    unsigned long iterate_addr = (unsigned long) READ_KERN(fops->iterate);
+    if (iterate_addr == 0 && iterate_shared_addr == 0)
+        return ;
+
+    unsigned long fops_addresses[3] ={
+        iterate_shared_addr,
+        (unsigned long) fops,
+        iterate_addr
+    };
+
+    save_u64_arr_to_buf(data, (const u64 *)fops_addresses, 3, 0);
+    events_perf_submit(data, DETECT_HOOKED_PROC_FOPS, 0);
+}
+
+SEC("kprobe/security_file_permission")
+int BPF_KPROBE(trace_security_file_permission)
+{
+    event_data_t data = {};
+    if (!init_event_data(&data, ctx))
+       return 0;
+
+    if (!should_trace(&data.context))
+        return 0;
+    struct file *f = (struct file *)PT_REGS_PARM1(ctx);
+    if (f == NULL)
+        return 0;
+
+    get_path_str(GET_FIELD_ADDR(f->f_path));
```

Comment:
@yanivagman, Itamar told me he couldn't use:

```C
buf_t *string_p = get_path_str(GET_FIELD_ADDR(f->f_path));
```

not sure the reason. Is there any specific reason he had to use get_buf() and get_buf_off() ?

Reply:
I need to see the verifier error first to try to answer that